packages/transformer/src/IModelExporter.ts

```diff
@@ -1145,6 +1205,132 @@ export class ChangedInstanceIds {
       this.handleChange(this.element, changeType, change.ECInstanceId);
   }
 
+  /**
+   * Adds the provided change to the appropriate set of changes by class type (codeSpec, model, element, aspect) maintained by this instance of ChangedInstanceIds.
+   * If the same ECInstanceId is seen multiple times, the changedInstanceIds will be modified accordingly, i.e. if an id 'x' was updated but now we see 'x' was deleted, we will remove 'x'
+   * from the set of updatedIds and add it to the set of deletedIds for the appropriate class type.
+   * @note This method will not accept changes to relationship classes. Use 'addCustomRelationshipChange' instead.
+   * @note In most cases, this method does not need to be called. Its only for consumers to mimic changes as if they were found in a changeset, which should only be useful in certain cases such as the changing of filter criteria for a preexisting master branch relationship.
+   * @throws if the ecClassId is a relationship classId
+   * @param ecClassId class id of the custom change
+   * @param changeType insert, update or delete
+   * @param id ECInstanceID of the custom change
+   * @param federationGuid federationGuid defined on the element
+   */
+  public async addCustomChange(
```

Comment:
- It would be great to set custom changes without passing ecClassId (and federation guid if possible).
- there is no indication if custom change is not added (e.g. class is not mapped). Suggest to throw error instead of silently ignoring invalid custom change. 
- Currently it is not registering custom changes for repository model, can be fixed by : 
```
 addECClassIdsToSet(this._modelSubclassIds, "BisCore.RepositoryModel"),
```

Reply:
Regarding the necessity of classId. It doesn't seem necessary, I am fine with trusting callers of the API to call the appropriate function. I converted addCustomChange to take in an EntityType instead of the classId and federationGuid. 
 
I do still wonder about relationships though and deleting them. Do you guys know whether or not you expect to have to add custom relationship changes? onDeleteRelationship seems to use the classFullName
 
Either way maybe we could just keep addCustomRelationshipChange as a slightly different API in that it would require classId if we had to...

Reply:
custom relationship changes would be added only in case if it is required when elements / models are marked as inserted/deleted. Is it needed in such cases or relationships will be handled automatically?

Reply:
Currently relationships will not be handled automatically and will need to be explicitly added to the sourceDbChanges, atleast inserts and updates it seems. Not sure about deletes (whether element deletes delete relationships automatically)

Im open to looking for relationships in addCustomElementChanges and adding those ids, but I do wonder if there would ever be a case where you just need to insert a relationship (relationship was filtered out/excluded?) 
 
or you wouldn't want relationship to come over just because you inserted the element.